core/domain/state_domain_test.py

```diff
@@ -1545,6 +1585,22 @@ def test_convert_html_fields_in_state_with_drag_and_drop_interaction(self):
             '"{&amp;quot;raw_latex&amp;quot;: &amp;quot;+,-,-,+&amp;quot;, &'
             'amp;quot;svg_filename&amp;quot;: &amp;quot;&amp;quot;}"></oppia'
             '-noninteractive-math>')
+
+        # We cannot use the above html strings as a test for customization args
```

Comment:
Is this going to cause problems for existing code which depends on the existing behaviour? This is an important question to answer correctly.

Reply:
I have a third option. Move the html cleaning into validate() instead of `__init__()`. Ensure that all paths to saving call validate() of SubtitledHtml before anything gets saved. In `__init__()`, also add a warning that this doesn't sanitize the HTML and the caller needs to ensure that validate() must be called on such objects prior to saving them to the datastore.

Reply:
(2) is not so good. It introduces a security hole. If you add that sort of argument then it is an invitation to other callers to use it in the future.

Reply:
Gotchu. I've updated the code to use option (3). I call validate on the SubtitledHtml where it is constructed (initializer or from_dict), and added a method for InteractionCustomizationArgs to validate all SubtitledHtml, which is only called in update_interaction_customization_args, and not in the html conversion function above.